src/rpc/client.zig

```diff
@@ -1,2338 +1,555 @@
+        var result = try Result(?types.Block).init(allocator);
+        errdefer result.deinit();
 
-        return try self.makeRequestWithJRpcResponse([]u64, json.Value, "getBlocksWithLimit", "1", params);
-    }
+        var params_builder = Request.ParamsBuilder.init(result.arena.allocator());
+        try params_builder.addArgument("{d}", slot);
+        try params_builder.addConfig(config);
 
-    pub fn getBlockTime(self: *Self, slot: u64) !jsonrpc.Response(?u64) {
-        var params = [1]u64{slot};
-        return try self.makeRequestWithJRpcResponse(?u64, []u64, "getBlockTime", "1", &params);
-    }
+        result.value = try self.sendFetchRequest(result.arena, ?types.Block, .{
+            .method = "getBlock",
+            .params = try params_builder.build(),
+        });
 
-    pub fn getClusterNodes(self: *Self) !jsonrpc.Response([]types.NodeInfo) {
-        return try self.makeRequestWithJRpcResponse([]types.NodeInfo, ?u8, "getClusterNodes", "1", null);
+        return result;
     }
 
-    const GetEpochInfoOptions = struct {
-        commitment: ?types.Commitment = null,
-    };
+    pub fn getBlockCommitment(self: *Client, allocator: std.mem.Allocator, block: u64) !Result(types.BlockCommitment) {
+        var result = try Result(types.BlockCommitment).init(allocator);
+        errdefer result.deinit();
 
-    pub fn getEpochInfo(self: *Self, options: GetEpochInfoOptions) !jsonrpc.Response(types.EpochInfo) {
-        var optionsObj = std.StringArrayHashMap(json.Value).init(self.allocator);
-        defer optionsObj.deinit();
-        try optionsObj.put("commitment", .{ .string = self.defaultCommitmentOr(options.commitment).string() });
+        var params_builder = Request.ParamsBuilder.init(result.arena.allocator());
+        try params_builder.addArgument("{d}", block);
 
-        var paramsList = std.ArrayList(json.Value).init(self.allocator);
-        defer paramsList.deinit();
-        try paramsList.append(.{ .object = optionsObj });
-        const params = json.Value{ .array = paramsList };
-        return try self.makeRequestWithJRpcResponse(types.EpochInfo, json.Value, "getEpochInfo", "1", params);
-    }
+        result.value = try self.sendFetchRequest(result.arena, types.BlockCommitment, .{
+            .method = "getBlockCommitment",
+            .params = try params_builder.build(),
+        });
 
-    pub fn getEpochSchedule(self: *Self) !jsonrpc.Response(types.EpochSchedule) {
-        return try self.makeRequestWithJRpcResponse(types.EpochSchedule, ?u8, "getEpochSchedule", "1", null);
+        return result;
     }
 
-    const GetFeeForMessageOptions = struct {
+    pub const GetBlockHeightConfig = struct {
         commitment: ?types.Commitment = null,
+        minContextSlot: ?u64 = null,
     };
 
-    pub fn getFeeForMessage(self: *Self, base64EncodedMessage: []const u8, options: GetFeeForMessageOptions) !jsonrpc.Response(types.MessageFeeInfo) {
```

Comment:
You deleted a lot from this file. You don't think it's worth trying to salvage it later? Do you think it would be easier to rewrite it all from scratch?

Reply:
the code will continue to exist in the commit history, if we need to reference it later - the code has changed significantly so am ok with starting from scratch tbh -- wdyt @yewman

Reply:
Yeah i think that makes sense, the code has also changed quite a lot and implementing the request logic is now quite trivial. The main effort in adding new methods will just be defining the return types in rpc/types.zig from the rpc docs